src/benchmarks/real-world/Roslyn/Program.cs

```diff
@@ -15,12 +17,16 @@ public static async Task<int> Main(string[] args)
         {
             await Setup();
 
-            return BenchmarkSwitcher
-                .FromAssembly(typeof(Program).Assembly)
-                .Run(args, RecommendedConfig.Create(
-                    artifactsPath: new DirectoryInfo(Path.Combine(Path.GetDirectoryName(typeof(Program).Assembly.Location), "BenchmarkDotNet.Artifacts")),
-                    mandatoryCategories: ImmutableHashSet.Create("Roslyn")))
-                .ToExitCode();
+            var summary = BenchmarkRunner.Run<StageBenchmarks>(
+                RecommendedConfig.Create(
+                               artifactsPath: new DirectoryInfo(Path.Combine(Path.GetDirectoryName(typeof(Program).Assembly.Location),
+                                                                             "BenchmarkDotNet.Artifacts")),
+                               mandatoryCategories: ImmutableHashSet.Create("Roslyn"))
+                           .With(Job.Default
```

Comment:
`.With` does not replace `Job` defined in `RecommendedConfig` but instead adds one more `Job` to the `Config`. So the benchmarks are actually executed twice.

Could you please extend `RecommendedConfig.Create` with an optional `Job` argument and use it instead of the recommended default if it was provided?

Reply:
Ah, OK, this explains the behavior I was seeing. Lemme follow up with that.